src/main/java/com/karankumar/bookproject/backend/service/BookService.java

```diff
@@ -46,33 +46,20 @@ public Book findById(Long id) {
 
     @Override
     public void save(Book book) {
-        if (book != null) {
-            if (book.getAuthor() != null) {
-                authorService.save(book.getAuthor());
-                LOGGER.log(Level.INFO, "Saving author: " + book.getAuthor());
-            } else {
-                LOGGER.log(Level.SEVERE, "Author is null");
-            }
-
-            if (book.getShelf() != null) {
-                if (book.getShelf().getShelfName() == null) {
-                    LOGGER.log(Level.SEVERE, "Shelf name should not be null");
-                } else {
-                    LOGGER.log(Level.INFO,
-                        "Shelf name (" + book.getShelf().getShelfName() + ") is not null");
-                }
-            } else {
-                LOGGER.log(Level.SEVERE, "Shelf is null");
-            }
-
-            LOGGER.log(Level.INFO, "Book repository count before: " + bookRepository.count());
-            bookRepository.save(book);
-            LOGGER.log(Level.INFO, "Book repository count after: " + bookRepository.count());
-
-            LOGGER.log(Level.INFO, book.getTitle() + " saved");
-        } else {
-            LOGGER.log(Level.SEVERE, "Null book");
+        if (book == null ) {
+            LOGGER.log(Level.SEVERE,"Cannot save the null book");
+            return;
+        }
+        if(book.getAuthor()==null){
+            LOGGER.log(Level.SEVERE,"Without author book cannot be saved.");
+            return;
+        }
+        if(book.getShelf() == null || book.getShelf().getShelfName() == null){
```

Comment:
As above, could you add a space after the `if` and a space after the closing parenthesis?